pkg/utils/parse_arf_result.go

```diff
@@ -126,7 +134,152 @@ func newRuleHashTable(dsDom *xmlquery.Node) nodeByIdHashTable {
 func newOcilQuestionTable(dsDom *xmlquery.Node) nodeByIdHashTable {
 	return newHashTableFromRootAndQuery(dsDom, "//ds:component/ocil:ocil", "//ocil:boolean_question")
 }
+func newStateHashTable(dsDom *xmlquery.Node) nodeByIdHashTable {
+	return newHashTableFromRootAndQuery(dsDom, "//ds:component/oval-def:oval_definitions/oval-def:states", "*")
+}
+
+func newObjHashTable(dsDom *xmlquery.Node) nodeByIdHashTable {
+	return newHashTableFromRootAndQuery(dsDom, "//ds:component/oval-def:oval_definitions/oval-def:objects", "*")
+}
```

Comment:
uhm... let's stay put to the e2e tests, I have the feeling these three operations will be very memory intensive.

Adding a big table of sates, objects and values from a 9Mb (as is the RCHOS4 datastream) file might result in high memory usage. If this is the case, we might want to increase our memory limits for the aggregator pod. I'm fine with using more memory since this is good for performance, and this is a one-shot operation (you don't do compliance checks every minute).

Reply:
Sounds good! I added one more table to lookup test for rules, the previous method had some corner cases, now it is fixed